DNN Platform/Modules/ResourceManager/Services/ItemsController.cs

```diff
@@ -364,6 +404,12 @@ private object GetItemViewModel(object item)
                 thumbnailUrl = thumbnailsManager.ThumbnailUrl(ActiveModule.ModuleID, file.FileId, 110, 110)
             };
         }
+
+        private ModuleInstanceContext GetModuleContext()
+        {
+            var moduleControl = ModuleControlFactory.CreateModuleControl(this.ActiveModule) as IModuleControl;
+            return new ModuleInstanceContext(moduleControl);
+        }
```

Comment:
DI guys @bdukes @ahoefling if you have a better idea of how to access EditUrl (which lives in PortalModuleBase) from a WebApi method, I am all ears on improving this :)

Reply:
I think there is something missing if we want to access the `EditUrl`. We may need to add an `IModuleNavigationManager` to manage routes at the module level. This control would be similar to the `INavigaitonManager` but would take a parameter of ModuleId.

Another option, I don't recommend but it would be a solution to get through referencing the deprecated `ModuleControlFactory`. You could inject the `ModuleControlPipeline` which is the new version of that class and access the same `CreateModuleControl()` method.

It could look something like this
```C#
public class ItemController : DnnApiController
{
    // omitted code

    private readonly IModuleControlPipeline modulePipeline;

    public ItemController(IModuleControlPipeline modulePipeline)
    {
        this.modulePipeline = modulePipeline;
    } 

    // omitted code

    private ModuleInstanceContext GetModuleContext()
    {
        IModuleControl moduleControl = modulePipeline.CreateModuleControl(this.ActiveModule);
        return new ModuleInstanceContext(moduleControl);
    }

    // omitted code
}
```

Reply:
Excellent, works like a charm and implemented in this PR.

Reply:
Glad this worked